src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -776,32 +777,37 @@ static void uct_rc_mlx5_ep_reset_qp(uct_rc_mlx5_ep_t *ep)
      */
     memset(&qp_attr, 0, sizeof(qp_attr));
     qp_attr.qp_state = IBV_QPS_RESET;
-    ret = ibv_cmd_modify_qp(txqp->qp, &qp_attr, IBV_QP_STATE, &cmd, sizeof(cmd));
+    ret = ibv_cmd_modify_qp(qp, &qp_attr, IBV_QP_STATE, &cmd, sizeof(cmd));
     if (ret) {
-        ucs_warn("modify qp 0x%x to RESET failed: %m", txqp->qp->qp_num);
+        ucs_warn("modify qp 0x%x to RESET failed: %m", qp->qp_num);
     }
 #else
-    (void)uct_rc_modify_qp(txqp, IBV_QPS_ERR);
+    (void)uct_ib_modify_qp(qp, IBV_QPS_ERR);
 #endif
-}
 
-static UCS_CLASS_CLEANUP_FUNC(uct_rc_mlx5_ep_t)
-{
-    uct_rc_mlx5_iface_t *iface = ucs_derived_of(self->super.super.super.iface,
-                                                uct_rc_mlx5_iface_t);
-
-    uct_ib_mlx5_txwq_cleanup(&self->tx.wq);
-
-    uct_rc_mlx5_ep_reset_qp(self);
     uct_rc_mlx5_iface_commom_clean_srq(&iface->mlx5_common, &iface->super,
-                                       self->qp_num);
+                                       qp->qp_num);
 
     /* Synchronize CQ index with the driver, since it would remove pending
      * completions for this QP (both send and receive) during ibv_destroy_qp().
      */
     uct_rc_mlx5_iface_common_update_cqs_ci(&iface->mlx5_common, &iface->super.super);
-    (void)uct_rc_modify_qp(&self->super.txqp, IBV_QPS_RESET);
+    (void)uct_ib_modify_qp(qp, IBV_QPS_RESET);
```

Comment:
why need it second time?

Reply:
it's because we want to set the correct ci for the driver and trigger the call to mlx5_cq_clean(), then fetch the updated ci from the driver